src/Symfony/Component/Security/Core/Authentication/Provider/SimpleAuthenticationProvider.php

```diff
@@ -50,20 +48,7 @@ public function authenticate(TokenInterface $token)
         $user = $authToken->getUser();
 
         if (!$user instanceof UserInterface) {
-            try {
-                $user = $this->userProvider->loadUserByUsername($user);
```

Comment:
this is the currently not covered case which does not exist on 2.7. `loadUserByUsername()` should throw when no user can be found for the passed username